flink-core/src/main/java/org/apache/flink/types/parser/BigDecParser.java

```diff
@@ -55,9 +56,20 @@ public int parseField(byte[] bytes, int startPos, int limit, byte[] delimiter, B
 			return -1;
 		}
 
-		String str = new String(bytes, startPos, i - startPos);
 		try {
-			this.result = new BigDecimal(str);
+			final int length = i - startPos;
+			if (reuse == null || reuse.length != length) {
+				reuse = new char[length];
+			}
+			for (int j = 0; j < length; j++) {
+				final byte b = bytes[startPos + j];
+				if ((b < '0' || b > '9') && b != '-' && b != '+' && b != '.' && b != 'E' && b != 'e') {
+					throw new NumberFormatException();
+				}
+				reuse[j] = (char) bytes[startPos + j];
+			}
+
+			this.result = new BigDecimal(reuse);
```

Comment:
we can use the `BigDecimal(char[] in, int offset, int len)` and reuse the `reuse` object also if it is larger than then provided input.